Reimplement the now missing `curwin_col_off` FFI function binding, fix #38

bbjornstad opened the pull request:
Please see the resolved issue for more information about the specific issue this is solving. TLDR: removed a function binding which was necessary for plugin operation and reimplemented it according to @cassava's proposed solution.

This has worked for me for some time and I believe for other people as well.

bbjornstad committed:
fix(ffi): redefine win_col_off behavior to replace removed function, fixes #38

As noted by @cassava, as of a recent version of this plugin, the
`curwin_col_off` function was removed, leaving us to fend for ourselves.
Luckily, @cassava also proposed a fix, the likes of which seems to work on my
machine and from the limited reports, other individuals' machines as well.

This commit simply implements the patch specified by @cassava as the solution to
issues related to `curwin_col_off` no longer existing.

bbjornstad committed:
fix(fill): adjust fill expansion computation to account for deprecations

Not only was the original function `curwin_col_off` deprecated and removed, but
we also have a number of recent deprecations related to vim operations on
tables. In this case, we need to get rid of calls to `tbl_flatten` and replace
them with `vim.iter():flatten():totable()` wrappings

bbjornstad committed:
chore(format): minor reformat on `init.lua` for plugin

bbjornstad committed:
Merge pull request #1 from bbjornstad/ffi_curwin_fix

Ffi curwin fix

bbjornstad committed:
fix(ffi): correctly include variant definition for `curwin_col_off`

I accidentally forgot this part in the first commit, but we do still need a
function called `curwin_col_off` even though it was not used directly in
`init.lua`.

bbjornstad committed:
Merge pull request #2 from bbjornstad/ffi_curwin_fix_hotfix

fix(ffi): correctly include variant definition for `curwin_col_off`

bbjornstad commented:
If anyone finds this hoping that this issue has been fixed in the upstream version of pretty-fold.nvim, until this PR is merged you can substitute my fork in your configuration ([bbjornstad/pretty-fold.nvim](https://github.com/bbjornstad/pretty-fold.nvim)). When this makes its way into the original repository I will remove my fork and one can move back to the upstream repo.

taoso commented:
@bbjornstad I am using you fork, it works like charming 👍

daUnknownCoder commented:
> If anyone finds this hoping that this issue has been fixed in the upstream version of pretty-fold.nvim, until this PR is merged you can substitute my fork in your configuration ([bbjornstad/pretty-fold.nvim](https://github.com/bbjornstad/pretty-fold.nvim)). When this makes its way into the original repository I will remove my fork and one can move back to the upstream repo.

might want to add this fork to rockerboo/awesome-neovim, because the last commit to this one was 2 years ago

Commit:
fix(components): swap tbl_flatten

    - tbl_flatten is deprecated and will be removed in Nvim 0.13
    - This patch exchanges the affected parts with the suggested
      alternative vim.iter(…):flatten():totable()

bbjornstad committed:
Merge pull request #3 from Fischmiep/master

fix(components): swap tbl_flatten
